### PR TITLE
Jakarta EE 9 GlassFish 6 Support

### PR DESCRIPTION
**Pull request by juneau001**
Adds support for GlassFish 6 automatic registration; Prepares for Jakarta EE 9.1 and GlassFish 6.1.0, although not yet enabled for automatic registration.

### CODE REVIEW

File: enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            </fileset>
+            <fileset dir="modules/endorsed">
+                <include name=".+\.jar"/>
+            </fileset>
+            <fileset dir="../mq/lib">
+                <include name="jaxm-api.jar"/>
+            </fileset>
+        </classpath>
+        <javadocs>
+            <lookup path="docs/javaee-doc-api.jar"/>
```
Comment (pepness) [inline enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml:73]: I think this should be `jakartaee9-doc-api.jar` or should we create a new `jakartaee91.platform` and use `jakartaee91-doc-api.jar`?

File: enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            <fileset dir="modules">
+                <include name="guava.+\.jar"/>
+                <include name="jackson.+\.jar"/>
+                <!-- include name="javax.ws.+\.jar"/ -->
+                <include name="jersey.+\.jar"/>
+                <include name="jettison.*\.jar"/>
+            </fileset>
+        </classpath>
+        <javadocs>
+            <link url="https://repo1.maven.org/maven2/org/glassfish/jersey/jersey-documentation/2.0-m05-2/jersey-documentation-2.0-m05-2-docbook.zip"/>
```
Comment (pepness) [inline enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml:89]: Jersey javadocs should be:
`https://repo1.maven.org/maven2/org/glassfish/jersey/jersey-documentation/3.0.1/jersey-documentation-3.0.1-docbook.zip`

File: enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml (hunk 1 above)
Comment (juneau001) [reply enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml:73]: Thanks @pepness, I will make the change for 12.4.  I was planning to make a new jakartaee91.platform once things are available, but just wanted to get a head-start on it for this release.

File: enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml (hunk 2 above)
Comment (juneau001) [reply enterprise/glassfish.tooling/src/org/netbeans/modules/glassfish/tooling/server/config/GlassFishV6_1_0.xml:89]: Thanks

**Comment by juneau001** [comment]
I see that the Travis build failed, but it is difficult to tell exactly what caused the failure.  The logs indicate the following, but I have not modified anything to my recollection which would have caused any issues with the hints:

 [junit] Test org.netbeans.modules.java.hints.errors.AccessErrorTest FAILED

Any ideas how I can move forward with this PR?  Thanks in advance!

**Comment by ebarboni** [comment]
seems tests are OK would it me mergeable

**Comment by emaphis** [comment]
If it matters, Glassfish 6.2 now exists.
https://github.com/eclipse-ee4j/glassfish/tags
Glassfish 6.2 includes Jakarta MVC

**Comment by juneau001** [comment]
Thanks, yes I am working on support for Glassfish 6.2 and Jakarta EE 9.1.

**Comment by ebarboni** [comment]
@juneau001 file are in conflict would it be possible to fix so we may merge ?

**Comment by juneau001** [comment]
> @juneau001 file are in conflict would it be possible to fix so we may merge ?

I will try to repair the conflict.  Thanks

**Comment by richard-grin** [comment]
I tried to declare a Payara 6 server in NetBeans 12.4 but it didn't work: a 6.2021.1 install is detected but when I click on Next, nothing happens (no error message) and "Finish" is greyed.
Do you plan to add a support for Payara 6 in the next 12.5 version?

**Comment by juneau001** [comment]
Hi @richard-grin Payara usually has their own support in Apache NetBeans.  I suspect that they will be adding support for Payara 6 in a future update.  This particular fix will only cover GlassFish 6.  Thanks for your feedback.

**Comment by richard-grin** [comment]
Thanks Josh for the information.

Le 19/08/2021 à 04:56, Josh Juneau a écrit :

**Pull request merged**